src/components/auth/auth.service.ts

```diff
@@ -184,6 +187,92 @@ export class AuthService {
     return session;
   }
 
+  async forget(email: string): Promise<boolean> {
+    const token = this.encodeJWT();
+
+    await this.db
+      .query()
+      .raw(
+        `
+      CREATE(et:EmailToken{value:$value, token: $token, createdOn:datetime()})
+      RETURN et as emailToken
+      `,
+        {
+          value: email,
+          token: token,
+        }
+      )
+      .first();
+    const params = {
+      Destination: { ToAddresses: ['leopard3551@gmail.com'] },
+      Message: {
+        Body: {
+          Html: {
+            Charset: 'UTF-8',
+            Data: `<html><body><p>This is your secret login code:</p>
+                          <a href="${this.config.resetPasswordURL}?token=${token}">Go to Login</a></body></html>`,
+          },
+          Text: {
+            Charset: 'UTF-8',
+            Data: `${this.config.resetPasswordURL}?token=${token}`,
+          },
+        },
+        Subject: {
+          Charset: 'UTF-8',
+          Data: 'Forget Password',
+        },
+      },
+      Source: this.config.emailForm,
+    };
+
+    await this.ses.sendEmail(params).promise();
+
+    return true;
+  }
+
+  async reset(input: ResetPasswordInput): Promise<boolean> {
+    const { token, password } = input;
+    const checkDate = new Date();
+
+    const result = await this.db
+      .query()
+      .raw(
+        `
+        MATCH(emailToken: EmailToken{token: $token})
+        RETURN emailToken.value as email, emailToken.token as token, emailToken.createdOn as createdOn
+        `,
+        {
+          token: token,
+        }
+      )
+      .first();
+    if (result) {
+      if (
+        Math.abs(
+          (checkDate.getTime() - Date.parse(result.createdOn)) / (1000 * 3600)
+        ) > 24
+      ) {
+        return false;
+      }
+      await this.db
+        .query()
+        .raw(
+          `
+          MATCH(e:EmailToken{token: $token})
+          DELETE e WITH * OPTIONAL MATCH(:EmailAddress{active: true, value:"test@test.com"})<-[:email{active:true}]-(user:User{active:true})-[:password{active:true}]->(password:Property{active:true})
+          SET password.value=$password return password
```

Comment:
Looks like the email address is hardcoded here. Shouldn't it be grabbed from somewhere?